### PR TITLE
feat: added primaryKeyComposite 

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: src/modules/model.js [hunk 1]
```diff
@@ -78,23 +78,43 @@ class Model {
   }
 
   // eslint-disable-next-line no-underscore-dangle
-  _generateId(data, providedId) {
-    assert(data instanceof Object && !Array.isArray(data), data);
-    assert(providedId === null || typeof providedId === 'string');
-    if ((typeof providedId !== 'string') === (!Array.isArray(this.primaryKeys))) {
-      throw new MustProvideIdXorPrimaryKeys();
+  _generateId(id, modelData = null) {
```
Comment (simlu) [inline src/modules/model.js]: could probably use some more assertions...

File: src/modules/model.js [hunk 2]
```diff
@@ -78,23 +78,43 @@ class Model {
...
-    if (Array.isArray(this.primaryKeys) && this.primaryKeys.some((k) => data[k] === undefined)) {
-      throw new IncompletePrimaryKey();
+    const toCheck = [id, modelData];
+    for (let idx = 0; idx < toCheck.length; idx += 1) {
+      const data = toCheck[idx];
+      if (
+        data instanceof Object
+        && !Array.isArray(data)
+        && this.primaryKeys !== null
+        && this.primaryKeys.every((k) => data[k] !== undefined)
```
Comment (simlu) [inline src/modules/model.js]: Should be any (!) and revert test!

File: src/modules/model.js [hunk 3]
```diff
@@ -78,23 +78,43 @@ class Model {
...
-      throw new MustProvideIdXorPrimaryKeys();
+  _generateId(id, modelData = null) {
+    let result = null;
+    if (typeof id === 'string') {
+      result = id;
     }
-    if (Array.isArray(this.primaryKeys) && this.primaryKeys.some((k) => data[k] === undefined)) {
-      throw new IncompletePrimaryKey();
+    const toCheck = [id, modelData];
+    for (let idx = 0; idx < toCheck.length; idx += 1) {
```
Comment (MacMcIrish) [inline src/modules/model.js]: Why not `toCheck.forEach`?

File: src/modules/model.js (hunk 2 above)
Comment (simlu) [reply src/modules/model.js]: Not necessary since there is a check that primary keys are not updated

File: src/modules/model.js (hunk 3 above)
Comment (MacMcIrish) [reply src/modules/model.js]:
```suggestion
    [id, modelData].forEach((data) => {
```

**Comment by simlu** [comment]
Has a whole bunch of other problems...

**Pull request closed without merging**